debugging.md

```diff
@@ -4,61 +4,70 @@ This page details information on common problems you might have with a command,
     * Or open and close the command block. If the time on the [previous output message](http://i.imgur.com/k2rmrXS.png) is not updating, the command block is not running
     * If there is no previous output, the command may have never ran, or you have its output turned off. Turn the command block's output [on](http://i.imgur.com/s4DYa9L.png) when you are debugging
     * You can fix this by turning the command block to *Needs Redstone*, pressing *Done*, turning it back to *Always Active*, then pressing *Done* again 
 * Check for double &nbsp;spaces between arguments, especially after copy pasting a part of the command. These are ignored in chat, but not elsewhere
+* Check for a space at the end of the command, command should **not** end with a space
 * Don't miss out arguments. Common ones to forget are:
   * The coordinates before the NBT data in `/summon`
   * The coordinates before the command in `/execute`
   * The old block handling mode (replace, keep, etc.) before NBT data in `/setblock` or `/fill`
-  * The block state/datavalue in `/setblock`, `/execute ~ ~ ~ detect` or `/fill`
+  * The block state/datavalue in `/setblock`, `/execute if block` or `/fill`
 * Watch out for `“smart quotes”` that word processors might auto-add, only `"normal quotes"` will work. You should use a plain plain text editor (Notepad, Notepad++, Sublime, Code), **not** a word processor or rich text editor (Microsoft Word, Wordpad, Textedit)
 * Narrow down your problem as much as possible. Remove parts slowly (or build up your command slowly in the first place) until you have just the part that's causing the issue
 * Macs add weird characters that are invisible in-game when the arrow keys are pressed. These will stop the command from working
 * Mods/plugins (especially Essentials) may overwrite vanilla commands. Try `/minecraft:command` instead of `/command` for the vanilla implementation (e.g: `/minecraft:give`)
     * If you have any mods, try vanilla to see if the mod is causing the problem. Even mods like Optifine can cause issues
+* Make sure command blocks are enabled in server.propieties
+* When using tutorials or [online generators](wiki/resources.md) make sure that are for the correct edition (java or bedrock) and version (1.12, 1.17, 1.20)
+* Check if the command block is in a loaded chunk, you can use the [`/forceload`](https://minecraft.wiki/w/Commands/forceload) command to force a chunk to be always loaded
 
 ## Functions
 
 * Make sure file extensions are not hidden ([Windows](http://i.imgur.com/FJ9x9Yg.png), [Mac](http://i.imgur.com/xDdYbXL.png)), otherwise a file that looks like `function.mcfunction` might actually be `function.mcfunction.txt`
 * [Check what errors you are receiving in the game log](https://i.imgur.com/vfwl8FX.png) (this isn't chat)
 * Commands cannot start with a `/` in functions (you'll get told this if you check the game log)
+* Macros lines **must** start with `$` and the values are specified using `$(example)`
```

Comment:
It is also worth mentioning that a macro command must contain at least one insert macro. Also inserting from a list/array or NBT path is not allowed.

Reply:
I edit it